lib/install/deps.js

```diff
@@ -254,11 +254,16 @@ exports.loadRequestedDeps = function (args, tree, saveToDependencies, log, next)
   }, andForEachChild(loadDeps, andFinishTracker(log, next)))
 }
 
+function isNotEmpty (value) {
+  return value != null && value !== ''
+}
+
 module.exports.computeVersionSpec = computeVersionSpec
 function computeVersionSpec (tree, child) {
   validate('OO', arguments)
   var requested
-  if (child.package._requested) {
+  var childReq = child.package._requested
+  if (childReq && (isNotEmpty(childReq.saveSpec) || (isNotEmpty(childReq.rawSpec) && isNotEmpty(childReq.fetchSpec)))) {
```

Comment:
We insist that `fetchSpec` exist with `rawSpec` because it  indicates modern metadata, where as `rawSpec` alone does not.